wordpress-mobile/WordPress-iOS
# Display Dynamic Cards in the Dashboard

Part of #22153
Fixes #22154
Fixes #22150

## Description

This PR displays dynamic cards in the dashboard.

| Light Mode | Dark Mode |
| ----------- | ------------ |
|  <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/febf8d8b-9371-45d9-aa3e-80b80fa21e72" /> | <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/6c1ccaa8-34ac-4144-b441-1b213c31c961" /> |

## Test Instructions

1. Install Jetpack and login.
2. **Expect** `New: Domains Management` is not shown.
3. Enable `Dynamic Dashboard Cards` Feature Flag.
4. Navigate to My Site.
5. **Expect** `New: Domains Management` card at the top.

## Regression Notes
1. Potential unintended areas of impact
The service responsible for fetching dashboard cards was refactored. However, the changes include only adding the dynamic cards to the existing list of cards. So technically the existing cards should work as expected. To be safe, it would be better to verify the existing cards are shown.

7. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

8. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
